### PR TITLE
Add `uv` based`pytest` workflow

### PR DESCRIPTION
**Pull request by DanielYang59**
Thanks a lot for the motivation and I just had a look at the `pymatviz` workflow setup which seems to point to this repo:

```yml
jobs:
  tests:
    strategy:
      matrix:
        os: [ubuntu-latest] # windows-latest
    uses: janosh/workflows/.github/workflows/pytest-release.yml@main
    with:
      os: ${{ matrix.os }}
      python-version: 3.9
    secrets: inherit
```

I guess I would need to add a `uv` based test workflow recipe?

Thanks a lot for giving me this chance to learn something new about GitHub workflow 😄

### CODE REVIEW

File: .github/workflows/pytest-release-with-uv.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        type: string
+        default: ""
+      install-uv:
+        description: Install uv
+        type: string
+        default: "pip install uv"
+      install-cmd-uv:
+        description: Install dependencies with uv
+        type: string
+        default: "uv pip install -e '.[test]'  --system"
```
Comment (DanielYang59) [inline .github/workflows/pytest-release-with-uv.yml]: This is what I added. All others are mostly just copied.

**Review by janosh — APPROVED** [review]
thank you!

File: .github/workflows/pytest-release-with-uv.yml (hunk 1 above)
Comment (janosh) [reply .github/workflows/pytest-release-with-uv.yml]: reusable GH workflows are a very nice feature. essentially means all my personal GH repos can be updated to use `uv` in CI by changing this 1 line 😄

Comment (DanielYang59) [reply .github/workflows/pytest-release-with-uv.yml]: Another new trick learned 😃

**Comment by DanielYang59** [comment]
Thought it might be safer to duplicate this, and redirect repos manually in case of any compatibility issue with `uv`. But I feel safe with you at my back 😄

**Comment by janosh** [comment]
my repos mostly use standard python dependencies which uv has been thoroughly tested against so feel pretty safe in making a complete migration. if not, you can always pin a repo to use the reusable workflow version before this PR

**Pull request merged**